agent/agent.go

```diff
@@ -265,57 +265,45 @@ func (a *Agent) runInputs(
 			interval = input.Config.Interval
 		}
 
+		var ticker Ticker
+		if a.Config.Agent.RoundInterval {
+			ticker = NewAlignedTicker(startTime, interval, jitter)
+		} else {
+			ticker = NewUnalignedTicker(interval, jitter)
```

Comment:
Just curious, when is unaligned useful?  I'm not sure I understand the issues we're fixing with this PR.

Reply:
For inputs this PR is more about the aligned ticker. The unaligned ticker is what is used when the agent has `round_interval = false` set and is essentially how the current tickers work.

The main property today that makes them useful is that they run immediately on startup.  With the aligned ticker you need to wait for the next aligned interval.  This is mostly an issue with large intervals when you don't want to wait for the initial collection.

Another way it can be used is with many systems running Telegraf to get similar results to what you would get with jitter.  Since each Telegraf will collect at different intervals they will be spread out in their requests to the outputs.  In some respects it is improved over jitter because you will always have the full `interval` between gathers, as opposed to with jitter when you have `interval - jitter`.

The other part of this PR is the rolling ticker as the only type for the outputs.  This is an ticker that isn't aligned and always has at least `interval` time even when using jitter.  Since outputs are less sensitive to the write time we are able to switch to this style with a uniform distribution over time.

Reply:
Ok, thanks for the context